lib/Backends/TraceEvents.cpp

```diff
@@ -121,6 +121,8 @@ ScopedTraceBlock::ScopedTraceBlock(TraceContext *context, llvm::StringRef name)
   if (context_) {
     context_->logTraceEvent(name_, TraceEvent::BeginType, std::move(args_));
   }
+
```

Comment:
I think that we need a comment here that explains what's going on and why do we have this fence.